x-pack/filebeat/input/websocket/config.go

```diff
@@ -0,0 +1,110 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package websocket
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type config struct {
+	// Program is the CEL program to be run for each polling.
+	Program string `config:"program"`
+	// Regexps is the set of regular expression to be made
+	// available to the program.
+	Regexps map[string]string `config:"regexp"`
+	// State is the initial state to be provided to the
+	// program. If it has a cursor field, that field will
+	// be overwritten by any stored cursor, but will be
+	// available if no stored cursor exists.
+	State map[string]interface{} `config:"state"`
+	// Auth is the authentication config for connection
+	Auth authConfig `config:"auth"`
+	// Resource
```

Comment:
Docs. Also, I'd rename this to URL (and in the configuration) since it can only ever be a URL (in CEL it can be a file, which is why it's called resource there).